### PR TITLE
Added methods that allow you to concatinate DataContainers together.

### PR DESCRIPTION
**Pull request by rmclaren**
Adds ability too merge data containers together via a .append method.

### CODE REVIEW

**Review by xincjin-NOAA — APPROVED** [review]
The changes look good to me.  

Nice work!

**Comment by emilyhcliu** [comment]
@rmclaren thanks for adding this capability.

File: test/testinput/bufrtest_python_test.py [hunk 1]
```diff
@@ -160,16 +160,35 @@ def test_highlevel_add():
...
+def test_highlevel_append():
+    DATA_PATH = 'testinput/data/gdas.t00z.1bhrs4.tm00.bufr_d'
+    YAML_PATH = 'testinput/bufrtest_hrs_basic_mapping.yaml'
+
+    container = bufr.Parser(DATA_PATH, YAML_PATH).parse()
+
+
+    new_container = bufr.DataContainer()
+    new_container.append(container)
+    new_container.append(container)
```
Comment (emilyhcliu) [inline test/testinput/bufrtest_python_test.py:172]: @rmclaren I did no check your PR in time.  I have one question.
Why do we needto append the container twice to the new_container?

**Review by emilyhcliu — COMMENTED** [review]
One question (see comment above).   Sorry about my late review.

File: test/testinput/bufrtest_python_test.py (hunk 1 above)
Comment (xincjin-NOAA) [reply test/testinput/bufrtest_python_test.py:172]: @emilyhcliu I guess this is testing appending one container into another one, though in the testcase these two containers are the same. The new_container is empty at the beginning.

Comment (rmclaren) [reply test/testinput/bufrtest_python_test.py:172]: It's just a test. So I start with an empty container, and then append the same data to 2 times. Later I check to make sure that is what actually happened (make sure appending the same data twice worked as expected).

**Pull request merged**